src/utils/Utils.ts

```diff
@@ -438,4 +438,16 @@ export class Utils {
   static resolveAfter<T = void>(ms: number, returns?: T): Promise<T> {
     return new Promise(resolve => setTimeout(() => (returns !== undefined ? resolve(returns) : resolve()), ms));
   }
+
+  static reorderValuesByKeys<T, K extends string>(values: T[], order: K[], getKey: (value: T) => K, appendRemainingValues = true) {
+    const valuesMap = values.reduce<{ [key: string]: T }>((map, value) => ({ ...map, [getKey(value)]: value }), {});
+    const orderedValues: T[] = [];
+    order.forEach(keyToAppend => {
+      if (valuesMap[keyToAppend]) {
+        orderedValues.push(valuesMap[keyToAppend]);
+        delete valuesMap[keyToAppend];
```

Comment:
Why is this `delete` statement needed?

Reply:
It's mostly a leftover from when I used `_.values` below. I left it since it serves a minor secondary purpose in preventing the value from being appended twice if `order` contains duplicate values. However, this isn't very important, I'll remove it.